wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java

```diff
@@ -274,22 +284,40 @@ public void read(HttpServiceRequest request, HttpServiceResponder responder,
   public void specification(HttpServiceRequest request, final HttpServiceResponder responder,
                             @PathParam("id") String id, @PathParam("topic") final String topic) {
     JsonObject response = new JsonObject();
+    String format = "text";
     try {
       Connection conn = store.get(id);
       JsonObject value = new JsonObject();
       JsonObject kafka = new JsonObject();
 
       Map<String, String> properties = new HashMap<>();
-      properties.put("topic", topic);
+      properties.put(PropertyIds.TOPIC, topic);
       properties.put("referenceName", topic);
       properties.put("brokers", conn.getProp(PropertyIds.BROKER));
       properties.put("kafkaBrokers", conn.getProp(PropertyIds.BROKER));
       properties.put("keyField", conn.getProp(PropertyIds.KEY_DESERIALIZER));
-      properties.put("format", "text");
+      String keytabLocation = conn.getAllProps().getOrDefault(PropertyIds.KEYTAB_LOCATION, runtimeArgs.get(PropertyIds.NAMESPACE_KETAB_PATH));
+      String principal = conn.getAllProps().getOrDefault(PropertyIds.PRINCIPAL, runtimeArgs.get(PropertyIds.NAMESPACE_PRINCIPAL_NAME));
+      if (keytabLocation != null && principal != null) {
+    	  properties.put(PropertyIds.KEYTAB_LOCATION, keytabLocation);
+    	  properties.put(PropertyIds.PRINCIPAL, principal);
+      }
 
-      kafka.add("properties", gson.toJsonTree(properties));
-      kafka.addProperty("name", "Kafka");
+      if (conn.getAllProps().containsKey(PropertyIds.KAFAK_PRODUCER_PROPERTIES)) {
+    	  Type type = new TypeToken<HashMap<String, String>>(){}.getType();
+    	  Map<String, String> kafkaProducerProperties = gson.fromJson(conn.getAllProps().get(PropertyIds.KAFAK_PRODUCER_PROPERTIES), type);
+    	  properties.put(PropertyIds.KAFKA_PROPERTIES, Joiner.on(",").withKeyValueSeparator(":").join(kafkaProducerProperties));
+    	  
+    	  if (kafkaProducerProperties.containsKey(PropertyIds.FORMAT) && Arrays.asList("text", "tsv", "binary", "avro", "csv").contains(kafkaProducerProperties.get(PropertyIds.FORMAT))) {
```

Comment:
add toLower 
kafkaProducerProperties.get(PropertyIds.FORMAT).toLower to make it case-insensitive if this value is not coming from any drop-down

Reply:
Adding lowercase